os/drivers/lwnl/lwnl.c

```diff
@@ -149,14 +141,14 @@ static int lwnl_close(struct file *filep)
 	int waiter_idx;
 #endif
 
+	LWNLDEV_LOCK(upper);
```

Comment:
Please check definition of -ENOSYS
It can't be used here. Also please see below.
```
if (upper->crefs > 1) {
     upper->crefs--;
} else {
     /* Last fd will be closed here, so we can free all resources, here and lower */
}
```